handlers/weniwebchat/weniwebchat.go

```diff
@@ -0,0 +1,259 @@
+		mediaURL = fmt.Sprintf("geo:%s,%s", payload.Message.Latitude, payload.Message.Longitude)
+	} else if payload.Message.MediaURL != "" {
+		mediaURL = payload.Message.MediaURL
+		payload.Message.Text = payload.Message.Caption
+	}
+
+	// build message
+	date := time.Unix(ts, 0).UTC()
+	msg := h.Backend().NewIncomingMsg(channel, urn, payload.Message.Text).WithReceivedOn(date).WithContactName(payload.From)
+
+	if mediaURL != "" {
+		msg.WithAttachment(mediaURL)
+	}
+
+	return handlers.WriteMsgsAndResponse(ctx, h, []courier.Msg{msg}, w, r)
+}
+
+var timestamp = ""
+
+type moPayload struct {
+	Type    string    `json:"type" validate:"required"`
+	To      string    `json:"to"   validate:"required"`
+	From    string    `json:"from" validate:"required"`
+	Message moMessage `json:"message"`
+}
+
+type moMessage struct {
+	Type         string   `json:"type"      validate:"required"`
+	TimeStamp    string   `json:"timestamp" validate:"required"`
+	Text         string   `json:"text,omitempty"`
+	MediaURL     string   `json:"media_url,omitempty"`
+	Caption      string   `json:"caption,omitempty"`
+	Latitude     string   `json:"latitude,omitempty"`
+	Longitude    string   `json:"longitude,omitempty"`
+	QuickReplies []string `json:"quick_replies,omitempty"`
+}
+
+func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStatus, error) {
+	start := time.Now()
+	status := h.Backend().NewMsgStatusForID(msg.Channel(), msg.ID(), courier.MsgSent)
+
+	baseURL := msg.Channel().StringConfigForKey(courier.ConfigBaseURL, "")
+	if baseURL == "" {
+		return nil, errors.New("blank base_url")
+	}
+
+	sendURL := fmt.Sprintf("%s/send", baseURL)
+
+	var logs []*courier.ChannelLog
+
+	payload := newOutgoingMessage("message", msg.URN().Path(), msg.Channel().Address(), msg.QuickReplies())
+	lenAttachments := len(msg.Attachments())
+	if lenAttachments > 0 {
+
+	attachmentsLoop:
+		for i, attachment := range msg.Attachments() {
+			mimeType, attachmentURL := handlers.SplitAttachment(attachment)
+			payload.Message.TimeStamp = getTimestamp()
+			// parse attachment type
+			if strings.HasPrefix(mimeType, "audio") {
+				payload.Message = moMessage{
+					Type:     "audio",
+					MediaURL: attachmentURL,
+				}
+			} else if strings.HasPrefix(mimeType, "application") {
+				payload.Message = moMessage{
+					Type:     "file",
+					MediaURL: attachmentURL,
+				}
+			} else if strings.HasPrefix(mimeType, "image") {
+				payload.Message = moMessage{
+					Type:     "image",
+					MediaURL: attachmentURL,
+				}
+			} else if strings.HasPrefix(mimeType, "video") {
+				payload.Message = moMessage{
+					Type:     "video",
+					MediaURL: attachmentURL,
+				}
+			} else {
+				elapsed := time.Since(start)
+				log := courier.NewChannelLogFromError("Error sending message", msg.Channel(), msg.ID(), elapsed, fmt.Errorf("unknown attachment mime type: %s", mimeType))
+				logs = append(logs, log)
+				status.SetStatus(courier.MsgFailed)
```

Comment:
we usually mark a message as errored so we can retry that up to 3 times, 
And in this case I think you want to just log the error as other parts could have been sent.